pandas/core/indexers.py

```diff
@@ -392,10 +390,10 @@ def check_array_indexer(array: AnyArrayLike, indexer: Any) -> Any:
 
     dtype = indexer.dtype
     if is_bool_dtype(dtype):
-        try:
+        if is_extension_array_dtype(dtype):
```

Comment:
we ought to think about adding a ``pd.asarray`` that does exactly this (as we have this idiom in several PRs).

cc @jorisvandenbossche @TomAugspurger

Reply:
In general that might be useful, but what would it solve here? (we are converting to numpy, not to an extension array, and we need to use a very specific na_value)